netsim/utils/templates.py

```diff
@@ -0,0 +1,141 @@
+#
+# Common routines for create-topology script
+#
+import sys
+import typing
+import os
+import pathlib
+
+from jinja2 import Environment, PackageLoader, FileSystemLoader, StrictUndefined, make_logging_undefined
+from box import Box,BoxList
+
+from .log import debug_active
+
+ansible_filter_map: dict = {}
+ANSIBLE_DEBUG = False
+
+#
+# Find path to the module directory (needed for various templates)
+#
+def get_moddir() -> pathlib.Path:
+  return pathlib.Path(__file__).resolve().parent.parent
+
+#
+# Find a file in a search path
+#
+def find_file(path: str, search_path: typing.List[str]) -> typing.Optional[str]:
+  for dirname in search_path:
+    candidate = os.path.join(dirname, path)
+    if os.path.exists(candidate):
+      return candidate
+
+  return None
+
+def template(j2: str , data: typing.Dict, path: str, user_template_path: typing.Optional[str] = None) -> str:
+  global ansible_filter_map
+  load_ansible_filters()
+
+  if path [0] in ('.','/'):                             # Absolute path or path relative to current directory?
+    template_path = [ path ]
+  else:                                                 # Path relative to netsim module, add module path to it
+    template_path = [ str(get_moddir() / path) ]
+  if not user_template_path is None:
+    template_path = [ './' + user_template_path, os.path.expanduser('~/.netlab/'+user_template_path) ] + template_path
+  if debug_active('template'):
+    print(f"TEMPLATE PATH for {j2}: {template_path}")
+  ENV = Environment(loader=FileSystemLoader(template_path), \
```

Comment:
Yeah, I thought about that as well, but the search paths are different, so I decided to stick with the environment generated on every call instead of trying to figure out how to deal with that.